ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -33,10 +33,21 @@ function(ament_target_dependencies target)
     message(FATAL_ERROR "ament_target_dependencies() the first argument must be a valid target name")
   endif()
   if(${ARGC} GREATER 0)
+    set(options PUBLIC)
```

Comment:
PRIVATE and INTERFACE keywords aren't allowed.
If someone tries to use them, this error will appear:
https://github.com/ament/ament_cmake/blob/664fea91b5659385b23a0669a9324a4142eb3e1d/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L53

Maybe, it's better to add them as an option and raise a clearer error.